packages/keystone/src/index.ts

```diff
@@ -0,0 +1,230 @@
+          '@ethereumjs/tx'
+        )
+
+        const { default: Common, Hardfork } = await import('@ethereumjs/common')
+
+        const keyring = AirGapedKeyring.getEmptyKeyring()
+        await keyring.readKeyring()
+
+        const eventEmitter = new EventEmitter()
+
+        let currentChain: Chain = chains[0]
+        const scanAccounts = async ({
+          derivationPath,
+          chainId,
+          asset
+        }: ScanAccountsOptions): Promise<Account[]> => {
+          currentChain =
+            chains.find(({ id }: Chain) => id === chainId) ?? currentChain
+
+          const provider = new providers.JsonRpcProvider(currentChain.rpcUrl)
+          return generateAccounts(keyring, provider)
+        }
+
+        const getAccounts = async () => {
+          accounts = await accountSelect({
+            basePaths,
+            assets,
+            chains,
+            scanAccounts,
+            walletIcon: await getIcon(),
+            supportsCustomPath: false
+          })
+
+          if (accounts.length) {
+            eventEmitter.emit('accountsChanged', [accounts[0].address])
+          }
+
+          return accounts
+        }
+
+        const keystoneProvider = {}
+        const provider = createEIP1193Provider(keystoneProvider, {
+          eth_requestAccounts: async () => {
+            // Triggers the account select modal if no accounts have been selected
+            const accounts = await getAccounts()
+            if (accounts?.length === 0) {
+              throw new ProviderRpcError({
+                code: 4001,
+                message: 'User rejected the request.'
+              })
+            }
+            return [accounts[0]?.address]
+          },
+          eth_accounts: async () => {
+            return accounts?.[0]?.address ? [accounts[0].address] : []
+          },
+          eth_chainId: async () => {
+            return currentChain?.id ?? ''
+          },
+          eth_signTransaction: async ({ params: [transactionObject] }) => {
+            if (!accounts)
+              throw new Error(
+                'No account selected. Must call eth_requestAccounts first.'
+              )
+
+            const account =
+              accounts.find(
+                account => account.address === transactionObject?.from
+              ) || accounts[0]
+
+            const { address: from } = account
+
+            // Set the `from` field to the currently selected account
+            transactionObject = { ...transactionObject, from }
+
+            const common = new Common({
+              chain: customNetwork || Number.parseInt(currentChain?.id) || 1,
+              // Berlin is the minimum hardfork that will allow for EIP1559
+              hardfork: Hardfork.Berlin,
+              // List of supported EIPS
+              eips: [1559]
+            })
+
+            transactionObject.gasLimit =
+              transactionObject.gas ?? transactionObject.gasLimit
```

Comment:
```suggestion
              transactionObject.gas || transactionObject.gasLimit
```

`gas` is either a string or is `undefined`